### ISSUE TITLE
Incompatible with Next.js 15 canary

### ISSUE THREAD

**Issue by denolfe** [opennextjs/opennextjs-cloudflare#61]
Hey All, we at [Payload](https://payloadcms.com) are extremely excited about getting this to work with our platform. People are asking about running Payload on Cloudflare quite often, and there is a lot of potential here.

Payload's version 3 is now built on Next.js but is still in beta and thus requires Next.js v15.0.0-canary.173 or higher. 

We followed the getting started docs [here](https://opennext.js.org/cloudflare/get-started), then attempted to upgrade to the latest Next.js + React versions and received the following errors:

```
/Users/elliot/dev/sandbox/my-next-app/node_modules/.pnpm/wrangler@3.80.0_@cloudflare+workers-types@4.20241004.0/node_modules/wrangler/wrangler-dist/cli.js:29768
            throw a;
            ^
{
  type: 'error',
  reason: 'Failed to construct initial bundle',
  cause: UserError: Failed to build
      at bundleWorker (/Users/elliot/dev/sandbox/my-next-app/node_modules/.pnpm/wrangler@3.80.0_@cloudflare+workers-types@4.20241004.0/node_modules/wrangler/wrangler-dist/cli.js:165378:15)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async build (/Users/elliot/dev/sandbox/my-next-app/node_modules/.pnpm/wrangler@3.80.0_@cloudflare+workers-types@4.20241004.0/node_modules/wrangler/wrangler-dist/cli.js:175368:59),
  source: 'BundlerController',
  data: undefined
}
```

We realize we're on the cutting edge here, but we're very excited to try this out. We'd like to know when we should try again.

**Comment by vicb** [comment]
Supporting v15 is definitely on our radar but not before the official release.

Our priority for now is to converge with Open Next/SST. It will enable moving faster in the future.

**Comment by sourabpramanik** [comment]
I am also trying to deploy the Payload 3.0 app which uses Next 15. I am able to build the app using wrangler successfully, just that when running the local server it throws errors.

Here are the error logs:
```
--- 2024-10-09T05:11:35.740Z debug
[InspectorProxyWorker] RUNTIME INCOMING MESSAGE {
  method: 'Runtime.consoleAPICalled',
  params: {
    type: 'error',
    args: [ [Object], [Object] ],
    executionContextId: 283096929,
    timestamp: 1728450695732,
    stackTrace: { callFrames: [Array] }
  }
}
---

--- 2024-10-09T05:11:42.559Z error
[31m✘ [41;31m[[41;97mERROR[41;31m][0m [1mFailed to prepare server Error: An error occurred while loading the instrumentation hook[0m


```

**Comment by eve0415** [comment]
Next.js v15 is now out!

https://github.com/vercel/next.js/releases/tag/v15.0.0

**Comment by nathanclevenger** [comment]
Is there an update or ETA for Next.js 15 support?

**Comment by vicb** [comment]
Probably second half of November

**Comment by shrinidhi4u** [comment]
Hi.. Any update on this?

**Comment by himanshu-ntml** [comment]
@denolfe, Opennext is compatible with Next.js 15. You can test it using the following prerelease package:

`pnpm add https://pkg.pr.new/@opennextjs/cloudflare@184`

**Comment by vicb** [comment]
☝️ when trying `https://pkg.pr.new/@opennextjs/cloudflare@184` be aware that it is on the experimental branch and the config has to be updated.

See https://github.com/opennextjs/opennextjs-cloudflare?tab=readme-ov-file#preleases and https://github.com/opennextjs/opennextjs-cloudflare/blob/experimental/packages/cloudflare/README.md

**Comment by vicb** [comment]
This should have been fixed in main / release 0.3.0.

Feel free to re-open if you still have issues.
